### ISSUE TITLE
[Beginner Question] Why Multiple Cores?

### ISSUE THREAD

**Issue by Marviel** [awslabs/aws-greengrass-mini-fulfillment#13]
Hi! Great project!

I'm a greengrass newbie trying to get a hang of things in the ecosystem, to see if it might be able to help our org out.

One question that has been bugging me is why the choice was made to have multiple cores in this example? Why _this way_, vs using _just one core_, that facilitates communication between each individual device in the whole application?

Just looking to learn :)

Thanks again for releasing the code for this cool project!
-Luke

**Comment by brettf** [comment]
You pose a very good question, and the simple answer is "because the size of the demo was unknown when we started".

The additional detail here is that, when we were making the mini-fulfillment center demo for Greengrass' launch a year ago, we did not know yet how physically big the demonstration was going to be. Therefore we could not know if a single host (with one core) could be physically wired to all of the servos of both arms **and** the master/conveyor. 

Since the launch, the physicality of this demo has actually proven to be the main reason to **continue** to use multiple cores. The wiring distances and setup of this example is just simpler when the long runs are done via Ethernet and the short runs are done with the servo protocol serial wires. The modularity that came from multiple cores also helped when we were iterated upon the size and shape of the robot arms, conveyor belt length, box size, etc..

But in full disclosure, using multiple Cores and therefore multiple Greengrass Groups *has* proven to bring about its own set of challenges - but those challenges all reside in software - and software is much easier to iterate rapidly upon when compared to the physical world. 

Does that make sense?
